sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreter.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.scripting
+
+import org.apache.spark.sql.catalyst.analysis.UnresolvedIdentifier
+import org.apache.spark.sql.catalyst.parser.{CompoundBody, CompoundPlanStatement, SingleStatement}
+import org.apache.spark.sql.catalyst.plans.logical.{CreateVariable, DropVariable, LogicalPlan}
+import org.apache.spark.sql.catalyst.trees.Origin
+
+/**
+ * Trait for SQL Scripting interpreters.
+ */
+trait ProceduralLanguageInterpreter {
+  /**
+   * Build execution plan and return statements that need to be executed,
+   *  wrapped in the execution node.
+   * @param compound CompoundBody for which to build the plan.
+   * @return Iterator through collection of statements to be executed.
+   */
+  def buildExecutionPlan(compound: CompoundBody) : Iterator[CompoundStatementExec]
+}
+
+/**
+ * Concrete implementation of the interpreter for SQL scripting.
+ */
+case class SqlScriptingInterpreter() extends ProceduralLanguageInterpreter {
+  /** @inheritdoc */
+  override def buildExecutionPlan(compound: CompoundBody): Iterator[CompoundStatementExec] = {
+    transformTreeIntoExecutable(compound).asInstanceOf[CompoundBodyExec]
+  }
+
+  /**
+   * Fetch the name of the Create Variable plan.
+   * @param plan Plan to fetch the name from.
+   * @return Name of the variable.
+   */
+  private def getDeclareVarNameFromPlan(plan: LogicalPlan): Option[UnresolvedIdentifier] =
+    plan match {
+      case CreateVariable(name: UnresolvedIdentifier, _, _) => Some(name)
```

Comment:
to confirm: we do creat session variables when executing the `CompoundBody`, but we remove it at the end to pretend it's local variables?

Reply:
Correct, this is not a target behavior and I explained a plan for variable related statements in [this comment](https://github.com/apache/spark/pull/47026#discussion_r1648683461) before, but we did it like this for now to simplify the PR and have a functioning scrips as well for testing purposes.